File: chazm-api/src/main/java/runtimemodels/chazm/api/relation/Needs.java
package runtimemodels.chazm.api.relation;

import runtimemodels.chazm.api.Organization;
import runtimemodels.chazm.api.entity.Attribute;
import runtimemodels.chazm.api.entity.Role;

/**
 * The {@linkplain Needs} interface defines the needs relation, where a {@linkplain Role} needs an {@linkplain Attribute}, of an {@linkplain Organization}.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public interface Needs {
    /**
     * Returns the {@linkplain Role} of this {@linkplain Needs}.
     *
     * @return the {@linkplain Role} of this {@linkplain Needs}.
     */
    Role getRole();

    /**
     * Returns the {@linkplain Attribute} of this {@linkplain Needs}.
     *
     * @return the {@linkplain Attribute} of this {@linkplain Needs}.
     */
    Attribute getAttribute();
}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/relation/Needs.kt
package runtimemodels.chazm.api.relation

import runtimemodels.chazm.api.entity.Attribute
import runtimemodels.chazm.api.entity.Role
import runtimemodels.chazm.api.organization.Organization

/**
 * The [Needs] interface defines a needs relation, where a [Role] needs an [Attribute], of an [Organization].
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
interface Needs {
    /**
     * The [Role] of this [Needs].
     */
    val role: Role

    /**
     * The [Attribute] of this [Needs].
     */
    val attribute: Attribute
}
